.github/workflows/ci.yml

```diff
@@ -308,3 +293,82 @@ jobs:
+      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          files: cov-linux-unit/codecov.json
+          fail_ci_if_error: true
+          flags: unittests,linux
+      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          files: cov-linux-nightly-unit/codecov.json
+          fail_ci_if_error: true
+          flags: unittests,linux-nightly
+      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          files: cov-macos-unit/codecov.json
+          fail_ci_if_error: true
+          flags: unittests,macos
+      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          files: cov-linux-integration/codecov.json
+          fail_ci_if_error: true
+          flags: integration-tests,linux
+      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          files: cov-linux-nightly-integration/codecov.json
+          fail_ci_if_error: true
+          flags: integration-tests,linux-nightly
+      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          files: cov-macos-integration/codecov.json
+          fail_ci_if_error: true
+          flags: integration-tests,macos
+      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          files: cov-py-backward-compat/codecov.json
+          fail_ci_if_error: true
+          flags: pytests,backward-compatibility,linux
+      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          files: cov-py-db-migration/codecov.json
+          fail_ci_if_error: true
+          flags: pytests,db-migration,linux
+      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          files: cov-py-sanity-check/codecov.json
+          fail_ci_if_error: true
+          flags: pytests,sanity-checks,linux
+      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          files: cov-py-genesis-check/codecov.json
+          fail_ci_if_error: true
+          flags: pytests,genesis-check,linux
+      - uses: codecov/codecov-action@eaaf4bedf32dbdc6b720b63067d99c4d77d6047d
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          files: cov-py-upgradability/codecov.json
+          fail_ci_if_error: true
+          flags: pytests,upgradability,linux
```

Comment:
You can use a script or something to generate the steps and then do something like `${{ fromJson(generated_steps) }}`.

Reply:
Hmmmmmm seems like this’d require having two jobs, right? If so I’m thinking that, for the current situation, it’s more effort than it’d currently be worth. Especially considering we still want specific flags for each upload depending on which file exactly is uploaded (otherwise we could just have the codecov action upload them all)

I’ve opened https://github.com/codecov/codecov-action/issues/1176 upstream, and am thinking we can probably live with this amount of copy-paste until upstream looks at it at least? :)